tests/angularModuleSpec.js

```diff
@@ -422,3 +422,60 @@ describe('StateCtrl', function () {
         }
     });
 });
+
+describe('TaskCtl', function () {
+    var scope,adalServiceProvider, rootScope, controller,window;
+    var store = {};
+    //mock Application to allow us to inject our own dependencies
+    beforeEach(angular.mock.module('TestApplication'));
+
+    //mock the controller for the same reason and include $scope and $controller
+    beforeEach(angular.mock.inject(function (_adalAuthenticationService_, _$rootScope_, _$controller_, _$httpBackend_, _$q_, _$window_, _$route_, _$location_) {
+        adalServiceProvider = _adalAuthenticationService_;
+        rootScope = _$rootScope_;
+        controller = _$controller_;
+        window = _$window_;
+        //create an empty scope
+        scope = rootScope.$new();
+
+        spyOn(sessionStorage, 'getItem').andCallFake(function (key) {
+            return store[key];
+        });
+        spyOn(sessionStorage, 'setItem').andCallFake(function (key, value) {
+            store[key] = value;
+        });
+        spyOn(window, 'Date').andCallFake(function () {
+            return {
+                getTime: function () {
+                    return 1000;
+                },
+                toUTCString: function () {
+                    return "";
+                }
+            };
+        });
+    }));
+
+    afterEach(function () {
+        store = {};
+    });
+
+    it('checks if acquireTokenSuccess/acquireTokenFailure events are broadcasted in case of acquireToken', function () {
+        var resource = adalServiceProvider.config.loginResource;
+        var token = 'token123';
+        spyOn(rootScope, '$broadcast').andCallThrough();
+        scope.$on('adal:acquireTokenFailure', function (event, message) {
+            expect(message).toBe('User login is required');
```

Comment:
We do not check if the events are called or not. This test will pass even when adal:acquireTokenFailure and adal:acquireTokenSuccess events are not called. We should either assign value of "message" to a variable defined in outside scope and test the value of that variable outside of the event listener. Or we should use "haveBeenCalled()" api to ensure that the event was called.